### ISSUE TITLE
[Feature request] Fine tuning of application icon positioning - it's too close to window buttons

### ISSUE THREAD

**Issue by michaldybczak** [kotelnik/plasma-applet-active-window-control#103]
The widget works great in AUR version. However some small tweak is needed. I recently moved from system panel to latte-panel (v. 0.7) and decided I finally bring that issue up.

As far I understand there are 3 main widget areas: 1) global menus 2) window buttons 3) application/window icon and name. There are many settings to place them.

The problem is with window icon placement in certain position. When I move it next to window buttons it just touches them creating very bad visual effect:

http://imgur.com/pSdUa4J

We need to either to have ability to move it a bit away from the buttons (10 px space should do) or this must be tweaked by the widget (hardcoded?). Latter option may however create various result with different scale setting so it's better to have flexibility.

Also, it would be nice to be able to leave the icon on the farthest left side of the AWC widget, just like it is by default on un-maximized windows (so additional setting, icon on the left). Basically transition to the maximized state should be coherent with general UI settings but I can't get it right now with AWC, despite its many settings.

At the moment I'm forced to have it like here:

http://imgur.com/a/91tii

Still, I would like to get more UI consistent state, here is the same window bar in un-maximized state:

http://imgur.com/a/wOK4G

If settings allowed us to get more or less the same look, it would feel like well designed behavior. Now it's a bit lacking. Plasma itself lacks of this features and AWC is the only hope to have it. In fact, all of this should be part of Plasma, not those badly working system global menus... AWC is a blessing and I would like to be even better :).

**Comment by kotelnik** [comment]
Hi! Thanks for suggestions and images and praise! I've added spacing between icon and buttons for more consistent look - it is the same margin used between icon and text (somebody already suggested that if I recall correctly).

I've also added an option to button settings: "Buttons between icon and text". It should work as expected - please test and report issues. Again same margin was used as said above. So you can adjust it by setting "Icon and text spacing" in Appearance menu.

As for the last proposed visual (icon left-most, buttons next to icon, text centered): understood. I suppose that the only tweak I should do is center the text. AppMenu will not be centered, correct?

**Comment by michaldybczak** [comment]
Updated to new version and wow! So much better :D

Now setting: buttons-icon-title-menu looks as expected!

http://imgur.com/q9NN2IX

Also thanks to new setting now I can really move icon to the left and having UI more consistent look (so similar as in unmaximized and maximized state) icon-butons-title-menu

http://imgur.com/oEcgN4V

Now the only missing option would be to move title on the right of menu (just for sake of customization, I'm undecided if that's the best setting or the one showed above), that way icon and title would gain full independence: icon-buttons-menu-title. With long titles like on this github page, menu is pushed beyond other widgets like here:

http://imgur.com/a/AZZLD

But that can be remedied with some title limits so this is not an issue:

http://imgur.com/a/25nqi

However it would be nice to test menu first and title after it. In that way menu would be always by the buttons and it would be nice to have menu always in the same place (moving menu may be confusing). However, I am still **VERY HAPPY** with current state :)

Maybe in some feature versions of AWC icon will be shown in settings as its own thing? This would be more intuitive, because now its placement settings are scattered between buttons and menu, which for newbies in AWC may be difficult to understand. So basically I propose to have instead 3 areas (1. icon with title 2. menu 3. buttons) to have fully independent 4 areas (1. icon 2. menu 3. buttons 4. title - not necessary in that order). But there is no hurry, just an idea to improve UI settings of AWC.

I spend many hours testing settings till I finally got it. I know that many people have troubles to understand them. The situation is worsened if they have version that is not fully functional so the settings do nothing. I first tested version from kde site, it was mostly non functional, then I changed to manjaro repo version and it was much better, but only when I switched to AUR git version, suddenly options start working and then I understood what it is about. But that was a few months ago, so I hope this is improved or fixed. However I always recommend git version as the best one, because people asking me about settings also had bad luck with other versions and finally got what they wanted if they installed git version.

So in short git version is just pure awesomeness and now combined with newest latte dock - ability to set global menus and various panel/dock effects skyrocketed through the roof! :D
There is currently no better DE then Plasma to have global menus! Well, the only thing that should be improved is to get gtk and electron menus working (electron is already fixed upstream so I hope to have global menus in electron apps soon), but that's for others to figure it out.

Huge thanks! AWC is awesome! Couldn't imagine Plasma without it!
